core/templates/pages/story-viewer-page/story-viewer-page.component.spec.ts

```diff
@@ -0,0 +1,400 @@
+            4: 24,
+            5: 46
+          },
+          id: '0',
+          activity_type: 'exploration'
+        },
+        completed: true,
+        thumbnail_bg_color: '#fff',
+        thumbnail_filename: 'story.svg'
+      }, {
+        id: 'node_2',
+        title: 'Title 2',
+        description: 'Description 2',
+        destination_node_ids: [],
+        prerequisite_skill_ids: ['skill_1'],
+        acquired_skill_ids: ['skill_2'],
+        outline: 'Outline',
+        outline_is_finalized: false,
+        exploration_id: null,
+        exp_summary_dict: {
+          category: 'Welcome',
+          created_on_msec: 1564183471833.675,
+          community_owned: true,
+          thumbnail_bg_color: '#992a2b',
+          title: 'Welcome to Oppia! 2',
+          num_views: 14897,
+          tags: [],
+          last_updated_msec: 1571653541705.924,
+          human_readable_contributors_summary: {},
+          status: 'public',
+          language_code: 'en',
+          objective: "become familiar with Oppia's capabilities 2",
+          thumbnail_icon_url: '/subjects/Welcome.svg',
+          ratings: {
+            1: 1,
+            2: 1,
+            3: 3,
+            4: 24,
+            5: 46
+          },
+          id: '0',
+          activity_type: 'exploration'
+        },
+        completed: false,
+        thumbnail_bg_color: '#000',
+        thumbnail_filename: 'story.svg'
+      }],
+      story_title: 'Story Title 1',
+      story_description: 'Story Description 1',
+      topic_name: 'topic_1',
+    });
+  }));
+
+  it('should get path icon parameters after story data is loaded', function() {
+    spyOn(storyViewerBackendApiService, 'fetchStoryData').and.returnValue(
+      $q.resolve(storyPlaythrough));
+    ctrl.$onInit();
+    $rootScope.$apply();
+
+    expect(ctrl.pathIconParameters).toEqual([{
+      thumbnailIconUrl: 'thumbnail-url',
+      left: '225px',
+      top: '35px',
+      thumbnailBgColor: '#fff'
+    }, {
+      thumbnailIconUrl: 'thumbnail-url',
+      thumbnailBgColor: '#000'
+    }]);
+  });
+
+  it('should show warning when fetching story data fails', function() {
+    spyOn(alertsService, 'addWarning');
+    spyOn(storyViewerBackendApiService, 'fetchStoryData').and.returnValue(
+      $q.reject({
+        status: 404
+      }));
+    ctrl.$onInit();
+    $rootScope.$apply();
+
+    expect(alertsService.addWarning).toHaveBeenCalledWith(
+      'Failed to get dashboard data');
+    expect(ctrl.pathIconParameters).toEqual([]);
+  });
+
+  it('should get complete exploration url when clicking on svg element',
+    function() {
+      var node = storyNodeObjectFactory.createFromIdAndTitle(
+        '1', 'Story node title');
+      expect(ctrl.getExplorationUrl(node)).toBe(
+        '/explore/null?topic_url_fragment=topic_1&' +
+        'classroom_url_fragment=clasroom_1&story_url_fragment=story_1&' +
+        'node_id=1');
+    });
+
+  it('should get static image url', function() {
```

Comment:
I remember suggesting a different name for such a test in a different PR. Can you edit it as per that?

Reply:
Done.